include/core/shape.h

```diff
@@ -531,4 +533,139 @@ class CompositeShape : public Shape<dim>
+    : Shape<dim>(support_radius[0], position, orientation)
+    , weight(weight)
+    , nodes(nodes)
+    , support_radius(support_radius)
+    , basis_function(basis_function)
+  {
+    unsigned int number_of_nodes = weight.size();
+
+    Point<dim>     high_bounding_point = Point<dim>();
+    Point<dim>     low_bounding_point  = Point<dim>();
+    Point<dim>     bounding_box_center = Point<dim>();
+    Tensor<1, dim> half_lengths        = Tensor<1, dim>();
+    for (int d = 0; d < dim; d++)
+      {
+        high_bounding_point[d] = DBL_MIN;
+        low_bounding_point[d]  = DBL_MAX;
+        for (unsigned int i = 0; i < number_of_nodes; i++)
+          {
+            if (low_bounding_point[d] > nodes[i][d])
+              low_bounding_point[d] = nodes[i][d];
+            if (high_bounding_point[d] < nodes[i][d])
+              high_bounding_point[d] = nodes[i][d];
+          }
+        bounding_box_center[d] =
+          0.5 *
+          (low_bounding_point[d] + high_bounding_point[d]); // + position[d];
+        half_lengths[d] =
+          0.5 * (high_bounding_point[d] - low_bounding_point[d]);
+      }
+    bounding_box = std::make_shared<Rectangle<dim>>(half_lengths,
+                                                    bounding_box_center,
+                                                    Tensor<1, 3>());
+  }
+
+  double
+  value(const Point<dim> & evaluation_point,
+        const unsigned int component = 0) const override;
+
+  std::shared_ptr<Shape<dim>>
+  static_copy() const override;
+
+  double
+  displaced_volume(const double fluid_density) override;
+
+  double
+  // Inspired by Optimad Bitpit. https://github.com/optimad/bitpit
+  evaluate_basis_function(const unsigned int basis_function_id,
+                          const double       distance) const;
+
+  double
+  wendlandc2(double) const;
+  double
+  linear(double) const;
+  double
+  gauss90(double) const;
+  double
+  gauss95(double) const;
+  double
+  gauss99(double) const;
+  double
+  c1c0(double) const;
+  double
+  c2c0(double) const;
+  double
+  c0c1(double) const;
+  double
+  c1c1(double) const;
+  double
+  c2c1(double) const;
+  double
+  c0c2(double) const;
+  double
+  c1c2(double) const;
+  double
+  c2c2(double) const;
+
+  std::pair<std::string, int>
+  get_shape_name() override
+  {
+    return std::make_pair("rbf_shape", Shape<dim>::ShapeType::rbf_shape);
+  }
+
+  /**
+   * Class taken from Optimad Bitpit. https://github.com/optimad/bitpit
+   * @enum RBFBasisFunction
+   * @ingroup RBF
+   * @brief Enum class defining types of RBF kernel functions that could be used in bitpit::RBF class
+   */
+  enum class RBFBasisFunction
+  {
+    CUSTOM     = 0,
+    WENDLANDC2 = 1,
+    LINEAR     = 2,
+    GAUSS90    = 3,
+    GAUSS95    = 4,
+    GAUSS99    = 5,
+    C1C0       = 6,
+    C2C0       = 7,
+    C0C1       = 8,
+    C1C1       = 9,
+    C2C1       = 10,
+    C0C2       = 11,
+    C1C2       = 12,
+    C2C2       = 13,
```

Comment:
Enums are numbered from 0 by default, so providing these values is optional.

Reply:
Also, since you take the time specify them as enum class, always use the enum class itself and never integer values.

Reply:
Good to know! And yes I'll make the change